modules/nlu/src/views/full/intents/IntentNameModal.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, Callout, Classes, Dialog, FormGroup, InputGroup, Intent } from '@blueprintjs/core'
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+
+interface Props {
+  // Used for actions rename and duplicate
+  originalName?: string
+  action: 'create' | 'rename' | 'duplicate'
+  intentNames: string[]
+  isOpen: boolean
+  toggle: () => void
+  onCreateIntent: (name: string) => void
+  onRenameIntent: (targetIntent: string, name: string) => void
+  onDuplicateIntent: (intentNameToDuplicate: string, name: string) => void
+}
+
+export const sanitizeName = (text: string) =>
+  text
+    .toLowerCase()
+    .replace(/\s|\t|\n/g, '-')
+    .replace(/[^a-z0-9-_.]/g, '')
+
+const IntentNameModal: FC<Props> = props => {
+  const [name, setName] = useState()
+
+  useEffect(() => {
+    props.action === 'rename' ? setName(props.originalName) : setName('')
+  }, [props.isOpen])
+
+  const submit = async e => {
+    e.preventDefault()
+
+    if (props.action === 'create') {
+      props.onCreateIntent(name)
+    } else if (props.action === 'duplicate') {
+      props.onDuplicateIntent(props.originalName, name)
+    } else if (props.action === 'rename') {
+      props.onRenameIntent(props.originalName, name)
+    }
+
+    closeModal()
+  }
+
+  const closeModal = () => {
+    setName('')
+    props.toggle()
+  }
+
+  const isIdentical = props.action === 'rename' && props.originalName === name
+  const alreadyExists = !isIdentical && _.some(props.intentNames, n => n.toLowerCase() === name.toLowerCase())
+
+  let dialog: { icon: any; title: string } = { icon: 'add', title: 'Create Intent' }
+  if (props.action === 'duplicate') {
+    dialog = { icon: 'duplicate', title: 'Duplicate Intent' }
+  } else if (props.action === 'rename') {
+    dialog = { icon: 'edit', title: 'Rename Intent' }
+  }
+
+  return (
+    <Dialog isOpen={props.isOpen} onClose={closeModal} transitionDuration={0} {...dialog}>
+      <form onSubmit={submit}>
+        <div className={Classes.DIALOG_BODY}>
+          <FormGroup label="Intent Name" helperText={`It can only contain letters, numbers, underscores and hyphens.`}>
+            <InputGroup
+              id="input-intent-name"
+              tabIndex={1}
+              placeholder="Choose a name for your intent"
+              required={true}
```

Comment:
in case you didn't know, in react when boolean props don't have to be set  to true explicitly. Let's take `required` here for instance 
```tsx
<InputGroup required={true}/> === <InputGroup required/>
<InputGroup required={false}/> === <InputGroup/>
```

Reply:
Thanks for info, yes, I am new to React and Node.

Actually, this was copied from `FlowNameModal.tsx`, will create a global PR for `required={true}` soon :)

Reply:
:)